Gem::Ext::BuildError: ERROR: Failed to build gem native extension. (use of undeclared identifier 'EV_UDATA_SPECIFIC')

Hello, I'm trying to build io-event (1.0.9) for a Solidus project (https://github.com/solidusio/solidus/discussions/4583) but I get this output 

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /usr/local/lib/ruby/gems/3.1/gems/io-event-1.0.9/ext
/usr/local/bin/ruby31 -I /usr/local/lib/ruby/site_ruby/3.1 -r
./siteconf20220907-35220-ny28ez.rb extconf.rb
checking for rb_ext_ractor_safe()... yes
checking for &rb_fiber_transfer()... yes
checking for -luring... no
checking for sys/epoll.h... no
checking for sys/event.h... yes
checking for sys/eventfd.h... no
checking for rb_io_descriptor()... yes
checking for &rb_process_status_wait()... no
checking for rb_fiber_current()... yes
checking for &rb_fiber_raise()... yes
checking for ruby/io/buffer.h... yes
creating extconf.h
creating Makefile

current directory: /usr/local/lib/ruby/gems/3.1/gems/io-event-1.0.9/ext
make DESTDIR\= clean

current directory: /usr/local/lib/ruby/gems/3.1/gems/io-event-1.0.9/ext
make DESTDIR\=
compiling ./io/event/event.c
compiling ./io/event/selector/selector.c
compiling ./io/event/selector/kqueue.c
./io/event/selector/kqueue.c:193:50: error: use of undeclared identifier 'EV_UDATA_SPECIFIC'
        event.flags = EV_ADD | EV_ENABLE | EV_ONESHOT | EV_UDATA_SPECIFIC;
                                                        ^
./io/event/selector/kqueue.c:217:28: error: use of undeclared identifier 'EV_UDATA_SPECIFIC'
        event.flags = EV_DELETE | EV_UDATA_SPECIFIC;
                                  ^
./io/event/selector/kqueue.c:276:60: error: use of undeclared identifier 'EV_UDATA_SPECIFIC'
                kevents[count].flags = EV_ADD | EV_ENABLE | EV_ONESHOT | EV_UDATA_SPECIFIC;
                                                                         ^
./io/event/selector/kqueue.c:291:60: error: use of undeclared identifier 'EV_UDATA_SPECIFIC'
                kevents[count].flags = EV_ADD | EV_ENABLE | EV_ONESHOT | EV_UDATA_SPECIFIC;
                                                                         ^
./io/event/selector/kqueue.c:313:38: error: use of undeclared identifier 'EV_UDATA_SPECIFIC'
                kevents[count].flags = EV_DELETE | EV_UDATA_SPECIFIC;
                                                   ^
./io/event/selector/kqueue.c:321:38: error: use of undeclared identifier 'EV_UDATA_SPECIFIC'
                kevents[count].flags = EV_DELETE | EV_UDATA_SPECIFIC;
                                                   ^
./io/event/selector/kqueue.c:697:20: error: use of undeclared identifier 'EVFILT_USER'
                trigger.filter = EVFILT_USER;
                                 ^
./io/event/selector/kqueue.c:698:39: error: use of undeclared identifier 'EV_UDATA_SPECIFIC'
                trigger.flags = EV_ADD | EV_CLEAR | EV_UDATA_SPECIFIC;
                                                    ^
./io/event/selector/kqueue.c:699:20: error: use of undeclared identifier 'NOTE_TRIGGER'
                trigger.fflags = NOTE_TRIGGER;
                                 ^
9 errors generated.
*** Error 1 in /usr/local/lib/ruby/gems/3.1/gems/io-event-1.0.9/ext (Makefile:246 'kqueue.o':
@ cc -I. -I/usr/local/include/ruby-3.1/x86_64-...)

make failed, exit code 2

Gem files will remain installed in /usr/local/lib/ruby/gems/3.1/gems/io-event-1.0.9 for
inspection.
Results logged to
/usr/local/lib/ruby/gems/3.1/extensions/x86_64-openbsd/3.1/io-event-1.0.9/gem_make.out
```
My OS is OpenBSD 7.1 x64 and my ruby version is 3.1.2

--- a/ext/extconf.rb
+++ b/ext/extconf.rb
@@ -50,7 +50,9 @@ if have_header('sys/epoll.h')
 	$srcs << "io/event/selector/epoll.c"
 end
 
-if have_header('sys/event.h')
+# The order matters, because we MUST have EV_UDATA_SPECIFIC.
+# The `have_header` call is just to add the -D to the compiler flags.
+if have_const('EV_UDATA_SPECIFIC', 'sys/event.h') and have_header('sys/event.h')
 	$srcs << "io/event/selector/kqueue.c"
 end
 

